mongomock/collection.py

```diff
@@ -1649,6 +1649,30 @@ def _extend_collection(out_collection, field, expression):
                 if k == '$match':
                     out_collection = [doc for doc in out_collection
                                       if filter_applies(v, doc)]
+                elif k == '$lookup':
+                    for operator in ['let', 'pipeline']:
+                        if operator in stage['$lookup']:
+                            raise NotImplementedError(
+                                "Although %s is a valid lookup operator for the "
+                                "aggregation pipeline, it is currently not "
+                                "implemented in Mongomock." % operator)
+                    for operator in ['from', 'localField', 'foreignField', 'as']:
+                        if operator not in stage['$lookup']:
+                            raise OperationFailure(
+                                "must specify '%s' field for a $lookup" % operator)
+                        if not isinstance(stage['$lookup'][operator], str):
+                            raise OperationFailure('arguments to $lookup must be strings')
+                    foreign_name = stage['$lookup']['from']
+                    local_field = stage['$lookup']['localField']
+                    foreign_field = stage['$lookup']['foreignField']
+                    local_name = stage['$lookup']['as']
+                    foreign_collection = self.database.get_collection(foreign_name)
+                    for doc in out_collection:
+                        query = doc[local_field] if local_field in doc else None
```

Comment:
Note that `local_field` could contain a `.`. I suggest that for now you just raise a `NotImplementedError` if you see this so that we can get your code merged. If someone else cares about using `.`, they will send another PR.

Also it's the same for `local_name`.